util.rb

```diff
@@ -0,0 +1,11 @@
+# ・濁点、半濁点、拗音、カタカナを、ひらがな清音に正規化。
+#  例:が→か ぽ→ほ ヅ→つ ゃ→や
+# ・アルファベット、数字の全角を、半角に正規化
+# ・記号は除く
+CANONICALIZE_FROM = 'がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽゔぁぃぅぇぉゃゅょアイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホマミムメモヤユヨラリルレロワヲンガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴァィゥェォャュョＡＢＣＤＥＦＧＨＩＪＫＬＭＮＯＰＱＲＳＴＵＶＷＸＹＺａｂｃｄｅｆｇｈｉｊｋｌｍｎｏｐｑｒｓｔｕｖｗｘｙｚ０１２３４５６７８９'
+CANONICALIZE_TO   = 'かきくけこさしすせそたちつてとはひふへほはひふへほうあいうえおやゆよあいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをんかきくけこさしすせそたちつてとはひふへほはひふへほうあいうえおやゆよABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+def canonicalize_initial str
+  return nil if str.nil? || str.empty?
+  s = str.gsub(/[「」『』（）（〔〕【】〈〉［］“”‘’\(\)\{\}\[\]・＠＃＄％＊！？＋＝@#$%\*\!\?\+=]/, "")
```

Comment:
```suggestion
  s = str.gsub(/[「」『』（）〔〕【】〈〉［］“”‘’\(\)\{\}\[\]・＠＃＄％＊！？＋＝@#$%\*\!\?\+=]/, "")
```
`（`が重複していたので削除してみました

Reply:
（こちらの方は無視してください :bow: ）